src/main/java/org/springframework/cloud/deployer/spi/kubernetes/support/PropertyParserUtils.java

```diff
@@ -40,14 +41,14 @@
 		Map<String, String> mapValue = new HashMap<>();
 
 		if (StringUtils.hasText(stringPairs)) {
-			String[] pairs = stringPairs.split(",");
+			String[] pairs = stringPairs.split(",(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)", -1);
```

Comment:
This seems to do the trick, however, would you mind adding a human description of what this magic regex does? 😉